libdnf/base/goal.cpp

```diff
@@ -905,6 +936,62 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
     }
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GroupJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    comps::GroupQuery group_query(base);
+    comps::GroupQuery group_query_names(group_query);
+    group_query_names.filter_name(spec, sack::QueryCmp::IGLOB);
+    group_query.filter_groupid(spec, sack::QueryCmp::IGLOB);
+    group_query |= group_query_names;
+    if (group_query.empty()) {
+        auto problem = transaction.p_impl->report_not_found(GoalAction::INSTALL, spec, GoalJobSettings(), strict);
+        if (strict) {
+            return problem;
+        } else {
+            return GoalProblem::NO_PROBLEM;
+        }
+    }
+    GoalProblem packages_res = GoalProblem::NO_PROBLEM;
+    for (auto group : group_query) {
+        auto packages = group.get_packages();
+        if (!settings.with_optional) {
+            std::erase_if(packages, [](comps::Package p) { return p.get_type() == comps::PackageType::OPTIONAL; });
+        }
+        auto pkg_settings = GoalJobSettings();
+        pkg_settings.with_provides = false;
+        pkg_settings.with_filenames = false;
+        pkg_settings.nevra_forms.push_back(rpm::Nevra::Form::NAME);
+        for (const auto & pkg : packages) {
+            auto pkg_name = pkg.get_name();
+            auto pkg_condition = pkg.get_condition();
+            if (pkg_condition.empty()) {
+                auto [pkg_problem, pkg_queue] = add_install_to_goal(transaction, pkg_name, pkg_settings);
+                rpm_goal.add_group_installed(pkg_queue);
+                packages_res |= pkg_problem;
+            } else {
+                add_provide_install_to_goal(fmt::format("({} if {})", pkg_name, pkg_condition), pkg_settings);
```

Comment:
What about to move `add_provide_install_to_goal` to `if (!query.empty()) {` and then also create `else` to report that group package is missing or we can also check the condition and only in case that condition is met by any package then we can create the report. What do you think?

Reply:
I'm not sure about reporting a missing package. At this stage we do not know whether the condition is met for the transaction and reporting a missing package could be misleading in case the condition is not met.